workflow/controller/hooks.go

```diff
@@ -18,12 +19,15 @@ func (woc *wfOperationCtx) executeWfLifeCycleHook(ctx context.Context, tmplCtx *
 		if hookName == wfv1.ExitLifecycleEvent {
 			continue
 		}
+		hookNodeName := generateLifeHookNodeName(woc.wf.ObjectMeta.Name, string(hookName))
```

Comment:
This is not needed until inside the execute block?

Reply:
I use it on new line 24, two lines below this one as part of the error message.